ts/config.ts

```diff
@@ -93,4 +93,13 @@ var config = AWS.config.loadFromPath('/to/path');
 // test update allowing unknown keys
 AWS.config.update({
    fake: 'fake' 
-}, true);
\ No newline at end of file
+}, true);
+
+// Test constructing with a CredentialProviderChain
+var options = {
+    credentialProvider: new AWS.CredentialProviderChain([
+        () => new AWS.EC2MetadataCredentials()
+    ])
+};
+var s3 = new AWS.S3(options);
+var ses = new AWS.SES(options);
```

Comment:
@chrisradek I am not exactly sure what the extent of the expected test should be. I feel like this covers the intent of what the change now provides. Obviously, if this is wrong I'll need a little bit more guidance on what to do next.